cmd/demo/node.go

```diff
@@ -122,6 +125,17 @@ func importAccount(secret string) (*ewallet.Wallet, *ewallet.Account, error) {
 	return wallet, acc.(*ewallet.Account), err
 }
 
+func getOnChainBal(ctx context.Context, addressList ...wallet.Address) (balList []*big.Int, err error) {
+	for _, addr := range addressList {
+		bal, err := ethereumBackend.BalanceAt(ctx, ewallet.AsEthAddr(addr), nil)
+		if err != nil {
+			return nil, errors.Wrap(err, "querying on-chain balance")
+		}
+		balList = append(balList, bal)
+	}
```

Comment:
If you know a slice size in advance, `make` that slice and then set its entries. Use `append` only if you're dynamically growing a slice.

Reply:
Then you also don't need a named return.

Reply:
Addressed in this commit - 13f6ee6.